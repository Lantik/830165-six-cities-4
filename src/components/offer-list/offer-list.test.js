import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import OfferList from './offer-list.jsx';

const mock = {
  offers: [{id: 1}, {id: 2}, {id: 3}],
  className: `test-class-name`,

};

const mockComponent = () => {
  return (<div></div>);
};

it(`<OfferList/> rendered correctly`, () => {
  const {offers, className} = mock;
  const renderer = new ShallowRenderer();

  renderer.render(<OfferList
    offers={offers}
    onOfferHeaderClick={() => {}}
    className={className}
    offerComponent={mockComponent}
    onOfferCardMouseEnter={() => {}}
  />);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});
