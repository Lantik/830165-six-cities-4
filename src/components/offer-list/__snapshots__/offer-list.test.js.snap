// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferList/> rendered correctly 1`] = `
<div
  className="test-class-name"
>
  <mockComponent
    offer={
      Object {
        "id": 1,
      }
    }
    onMouseEnter={[Function]}
    onOfferHeaderClick={[Function]}
  />
  <mockComponent
    offer={
      Object {
        "id": 2,
      }
    }
    onMouseEnter={[Function]}
    onOfferHeaderClick={[Function]}
  />
  <mockComponent
    offer={
      Object {
        "id": 3,
      }
    }
    onMouseEnter={[Function]}
    onOfferHeaderClick={[Function]}
  />
</div>
`;
